packages/grid/x-data-grid/src/constants/gridClasses.ts

```diff
@@ -364,8 +364,13 @@ export interface GridClasses {
   toolbarFilterList: string;
   /**
    * Styles applied to both the cell and the column header if `showColumnRightBorder={true}`.
+   * Sets border color only.
    */
   withBorder: string;
+  /**
+   * Styles applied to both the cell and the column header if `showColumnRightBorder={true}`.
+   */
+  withRightBorder: string;
```

Comment:
Split it?

`withRightBorder` -> `cell--withRightBorder`
`withRightBorder` -> `columnHeader--withRightBorder`

Both will have the same styles. It's more to be able to discover their purpose only by reading the name.

Reply:
Done